feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/ui/specialfolders/SpecialFoldersContract.kt

```diff
@@ -0,0 +1,66 @@
+package app.k9mail.feature.account.setup.ui.specialfolders
+
+import app.k9mail.core.ui.compose.common.mvi.UnidirectionalViewModel
+import app.k9mail.feature.account.common.domain.entity.Folder
+import app.k9mail.feature.account.common.ui.loadingerror.LoadingErrorState
+
+interface SpecialFoldersContract {
+
+    interface ViewModel : UnidirectionalViewModel<State, Event, Effect>
+
+    interface FormUiModel {
+        fun event(event: FormEvent, formState: FormState): FormState
+        fun validate(formState: FormState): Boolean
+    }
+
+    data class State(
+        val formState: FormState = FormState(),
+
+        val isSuccess: Boolean = false,
+        override val error: Failure? = null,
+        override val isLoading: Boolean = true,
+    ) : LoadingErrorState<Failure>
+
+    data class FormState(
+        val archiveFolders: List<Folder> = emptyList(),
+        val draftsFolders: List<Folder> = emptyList(),
+        val sentFolders: List<Folder> = emptyList(),
+        val spamFolders: List<Folder> = emptyList(),
+        val trashFolders: List<Folder> = emptyList(),
+
+        val selectedArchiveFolder: Folder? = null,
+        val selectedDraftsFolder: Folder? = null,
+        val selectedSentFolder: Folder? = null,
+        val selectedSpamFolder: Folder? = null,
+        val selectedTrashFolder: Folder? = null,
```

Comment:
They are null in case there is no matching folder type available, which would fail the validation and present the screent to the user to select a folder. By default the input selection will show `Automatic (None)` but it's not selected yet. User needs to actively select it. Maybe there needs to be an empty state or a caption highlighting that an action is required.